deps/libcaption/src/utf8.c

```diff
@@ -51,7 +51,7 @@ size_t utf8_char_length(const utf8_char_t* c)
 int utf8_char_whitespace(const utf8_char_t* c)
 {
     // 0x7F is DEL
-    if (!c || (c[0] >= 0 && c[0] <= ' ') || c[0] == 0x7F) {
+    if (!c || ((signed char)c[0] >= 0 && c[0] <= ' ') || c[0] == 0x7F) {
```

Comment:
Assuming `(c[0] >= 0 && c[0] <= ' ')` was checking `c[0]` is within [0x00, 0x20], both `c[0]` should have the same type cast.
```suggestion
    if (!c || ((signed char)c[0] >= 0 && (signed char)c[0] <= ' ') || c[0] == 0x7F) {
```

However, cast to `unsigned char` would be also ok. I feel either is fine.w
```suggestion
    if (!c || (signed char)c[0] <= ' ' || c[0] == 0x7F) {
```

Reply:
Thanks for your advice! I found that the unsigned-to-signed cast is implementation-defined so I'm taking the second one as signed-to-unsigned cast is well-defined.

Reply:
i don't know nothin' about types but I strongly assume that `' '` expands to a value of type `char` (whichever signedness the compiler prefers), so the following would actually introduce a signed-unsigned comparison on *some* platforms:
```C
(signed char)c[0] <= ' '
```

also note that the error we get is not really about a signed-unsigned comparison, but about a useless comparison (an `unsigned char` will *always* be `>= 0`) and so `-Werror=type-limits` (implied by `-Werror -Wextra`) errors out.

Reply:
> i don't know nothin' about types but I strongly assume that `' '` expands to a value of type `char` (whichever signedness the compiler prefers), so the following would actually introduce a signed-unsigned comparison on _some_ platforms:
> 
> ```c
> (signed char)c[0] <= ' '
> ```
> 
> also note that the error we get is not really about a signed-unsigned comparison, but about a useless comparison (an `unsigned char` will _always_ be `>= 0`) and so `-Werror=type-limits` (implied by `-Werror -Wextra`) errors out.

I think that's just a typo. What @norihiro means is `(unsigned char)c[0] <= ' '` which I have already pushed to this PR.

Also the signed-unsigned comparison is not a problem here because standard requires integer operands go through *integer promotions*, which will convert both operands to `int` before comparison.